js/adapt-contrib-text.js

```diff
@@ -6,31 +6,39 @@ define(function(require) {
     var Text = ComponentView.extend({
 
         preRender: function() {
-            // Checks to see if the text should be reset on revisit
             this.checkIfResetOnRevisit();
         },
 
         postRender: function() {
             this.setReadyStatus();
 
-            // Check if instruction or title or body is set, otherwise force completion
-            var cssSelector = this.$('.component-instruction').length > 0
-                ? '.component-instruction'
-                : (this.$('.component-title').length > 0 
-                ? '.component-title' 
-                : (this.$('.component-body').length > 0 
-                ? '.component-body' 
-                : null));
+            this.setupInview();
+        },
+
+        setupInview: function() {
+            var selector = this.getInviewElementSelector();
 
-            if (!cssSelector) {
+            if (!selector) {
                 this.setCompletionStatus();
             } else {
-                this.model.set('cssSelector', cssSelector);
-                this.$(cssSelector).on('inview', _.bind(this.inview, this));
+                this.model.set('inviewElementSelector', selector);
+                this.$(selector).on('inview', _.bind(this.inview, this));
             }
         },
 
-        // Used to check if the text should reset on revisit
+        /**
+         * determines which element should be used for inview logic - body, instruction or title - and returns the selector for that element
+         */
+        getInviewElementSelector: function() {
+            if(this.$('.component-body').length > 0) return '.component-body';
```

Comment:
I guess the chances of a plugin injecting text are quite slim.  I'd +1 for checking the model properties instead.